### PR TITLE
[AMDGPU] Enable `AAAddressSpace` in `AMDGPUAttributor`

### PR DESCRIPTION
**Pull request by shiltian**

### CODE REVIEW

**Comment by shiltian** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/101593?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#101593** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/101593?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#101591** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/101591?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#101589** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/101589?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @shiltian and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: llvm/lib/Target/AMDGPU/AMDGPUAttributor.cpp [hunk 1]
```diff
@@ -1064,6 +1064,17 @@ static bool runImpl(Module &M, AnalysisGetter &AG, TargetMachine &TM) {
     } else if (CC == CallingConv::AMDGPU_KERNEL) {
       addPreloadKernArgHint(F, TM);
     }
+
+    for (auto &I : instructions(F)) {
+      if (auto *LI = dyn_cast<LoadInst>(&I)) {
```
Comment (shiltian) [inline llvm/lib/Target/AMDGPU/AMDGPUAttributor.cpp:1069]: `AAAddressSpace` for now only supports load and store instructions. Will add more later.

**Comment by ssahasra** [comment]
Looks good to me, but the PR is still marked as a work in progress.

**Comment by shiltian** [comment]
> Looks good to me, but the PR is still marked as a work in progress.

Yeah, there is a large file `CodeGen/AMDGPU/GlobalISel/flat-scratch-init.ll` to be updated.

**Review by ssahasra — APPROVED** [review]

**Comment by shiltian** [comment]
I think the test was a false failure. What I changed was the optimization pipeline for AMDGPU, but this test case doesn't have GPU involved at all.

**Pull request merged**